doc/Frontend/ideas.org

```diff
@@ -0,0 +1,58 @@
+* Syntax thoughts
+Initial ideas on syntax
+** Arrows
+- Arrow syntax is relatively simple
+
+- Without sugar all arrows will look like
+
+  #+BEGIN_SRC idris
+    f : (x : Int) -1-> (y : Int) -w-> C -(x+y)-> D
+  #+END_SRC
+-
+  + Note :: In the unicode extension ω can replace w
+  + Note :: the last arrow requires dependent useages
+
+- With sugar it would look something like this
+  #+BEGIN_SRC idris
+    f : (x : Int) -o (y : Int) -> C -(x+y)-> D
+  #+END_SRC
+  + We can see that -o stands for 1 useage, and -> stands for
+    unlimited useages
+  + Note :: In the unicode extension → can replace ->
+  + Note :: In the unicode extension  ⊸ can replace -o
```

Comment:
I like the sugar options - what's the default though? Why not the simple:

`f : 2 (x : 3 Int) -> Double`, `f : w (x : 2 Int -> y : 1 () -> IO ())`

Reply:
I'm assuming most would either pick -o or -> as the default.

The issue with this, is that there is no way to alias this approach, meaning we always have to write explicit numbers, when an arrow signifier would be better.

Though I will say, this allows toplevel functions
```idris
f : 2 Int
f = 3
```

Or something along these lines

Reply:
> The issue with this, is that there is no way to alias this approach, meaning we always have to write explicit numbers, when an arrow signifier would be better.

Hmm why - couldn't we still alias the arrows? e.g. `f : 2 (x -> Double)` (if `f` is linear in `x`, but we need `2` of `f`)? 

Honestly I think I prefer the explicit usages though, it's a few more characters but easier to read since the usages are always between the variable binder and the type.

Reply:
> > The issue with this, is that there is no way to alias this approach, meaning we always have to write explicit numbers, when an arrow signifier would be better.
> 
> Hmm why - couldn't we still alias the arrows? e.g. `f : 2 (x -> Double)` (if `f` is linear in `x`, but we need `2` of `f`)?
> 

My issue is that imagine, if we didn't have `Double` but instead `m A`. would this not be amigious with usage m?